extensions/amp-cxense-player/0.1/amp-cxense-player.js

```diff
@@ -0,0 +1,244 @@
+ * limitations under the License.
+ */
+
+import {CSS} from '../../../build/amp-cxense-player-0.1.css';
+import {getLengthNumeral, isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+import {addParamsToUrl} from '../../../src/url';
+import {createLoaderElement} from '../../../src/loader';
+import {getDataParamsFromAttributes} from '../../../src/dom';
+import {setStyles} from '../../../src/style';
+
+const cxDefaults = {
+  apiHost: 'https://api.widget.cx',
+  distEmbedHost: 'https://embed.widget.cx',
+  debugEmbedHost: 'https://stage-embed.widget.cx',
+  distEmbedApp: '/app/player/m4/dist/',
+  debugEmbedApp: '/app/player/m4/debug/',
+    // can't access the window.top.location.href from the iframe to enable sharing the top url
+  attrs: {
+    'share.enable': false,
+  },
+};
+
+class AmpCxense extends AMP.BaseElement {
+
+    /** @override */
+    preconnectCallback(onLayout) {
+      this.preconnect.url(cxDefaults.apiHost, onLayout);
+    }
+
+    /** @override */
+    isLayoutSupported(layout) {
+      return isLayoutSizeDefined(layout);
+    }
+
+    /** @override */
+    renderOutsideViewport() {
+      return false;
+    }
+
+    /** @override */
+    buildCallback() {
+      this.element.classList.add('amp-cxense-player');
+
+        /** @private @const {object} */
+      this.cxDefaults_ = JSON.parse(JSON.stringify(cxDefaults));
+        /** @private @const {object} */
+      this.attrs_ = this.getDataAttributes_();
+
+      this.cxDefaults_.embedHost = this.attrs_.debug
+            ? this.cxDefaults_.debugEmbedHost : this.cxDefaults_.distEmbedHost;
+      this.cxDefaults_.embedApp = this.attrs_.debug
+            ? this.cxDefaults_.debugEmbedApp : this.cxDefaults_.distEmbedApp;
+
+        /** @private @const {number} */
+      this.width_ = getLengthNumeral(this.element.getAttribute('width'));
+        /** @private @const {number} */
+      this.height_ = getLengthNumeral(this.element.getAttribute('height'));
+
+      if (!this.getPlaceholder()) {
+        this.buildImagePoster_();
+        this.loading_(true);
+      }
+    }
+
+    /** @override */
+    layoutCallback() {
+      const self = this;
+
+        /** @private @const {Element} */
+      this.iframe_ = this.element.ownerDocument.createElement('iframe');
+
+      this.iframe_.setAttribute('frameborder', '0');
+      this.iframe_.setAttribute('allowfullscreen', 'true');
+
+      this.iframe_.width = this.width_;
+      this.iframe_.height = this.height_;
+
+      this.iframe_.src = this.getIframeSrc_();
+      setStyles(this.iframe_, {display: 'none'});
+      this.element.appendChild(this.iframe_);
+
+        /** @private {boolean} */
+      this.mpfReady_ = false;
+
+        /** @private @const {!Promise} */
+      this.playerReadyPromise_ = new Promise(resolve => {
+            /** @private @const {function()} */
+        this.playerReadyResolver_ = iframe => {
+          self.mpfReady = true;
```

Comment:
Since you're using arrow functions, `this` will always equal the Element. `self` is unnecessary.